### PR TITLE
GHA Migration

### PR DESCRIPTION
**Pull request by robmonte**
Github Actions Migration and Engineering Improvements:

* Remove CircleCI config, add Github Actions workflows
* Add dependabot
* Improve acceptance test running
* Remove gox
* Update README

### CODE REVIEW

File: .github/workflows/jira.yaml [hunk 1]
```diff
@@ -1,72 +1,19 @@
...
-
-    - name: Reopen ticket
-      if: github.event.action == 'reopened' && steps.search.outputs.issue
-      uses: atlassian/gajira-transition@v2.0.1
-      with:
-        issue: ${{ steps.search.outputs.issue }}
-        transition: "Pending Triage"
+    uses: hashicorp/vault-workflows-common/.github/workflows/jira.yaml@main
+    # assuming you use Vault to get secrets
+    # if you use GitHub secrets, use secrets.XYZ instead of steps.secrets.outputs.XYZ
```
Comment (fairclothjm) [inline .github/workflows/jira.yaml:13]: Let's remove both of these comments. I had them in my alicloud auth example but the first line is incorrect.

File: Makefile [hunk 2]
```diff
@@ -48,4 +50,4 @@ fmt:
 proto:
 	protoc *.proto --go_out=plugins=grpc:.
 
-.PHONY: bin default generate test vet bootstrap fmt fmtcheck
\ No newline at end of file
+.PHONY: bin default generate test vet bootstrap fmt fmtcheck
```
Comment (fairclothjm) [inline Makefile:53]:
```suggestion
.PHONY: bin default generate test testacc vet bootstrap fmt fmtcheck
```

File: scripts/build.sh [hunk 3]
```diff
@@ -41,55 +38,27 @@ rm -f bin/*
...
-
-# The main method we need for building is in the cmd directory
-cd "${DIR}/cmd/${TOOL}"
-
 # Build!
 echo "==> Building..."
-gox \
-    -osarch="${XC_OSARCH}" \
+echo "Running:"
+echo "${GO_CMD} build -gcflags ${GCFLAGS} -ldflags -X github.com/hashicorp/${TOOL}/version.GitCommit=${GIT_COMMIT}${GIT_DIRTY} -o bin/${TOOL} -tags ${BUILD_TAGS} ."
```
Comment (fairclothjm) [inline scripts/build.sh:44]: Looks like we have 2 `go build` commands in this file

Comment (fairclothjm) [reply scripts/build.sh:44]: oh, nevermind this is just `echo`-ing the command

File: scripts/build.sh [hunk 4]
```diff
@@ -41,55 +38,27 @@ rm -f bin/*
...
+echo ""
+${GO_CMD} build \
+    -gcflags "${GCFLAGS}" \
     -ldflags "-X github.com/hashicorp/${TOOL}/version.GitCommit='${GIT_COMMIT}${GIT_DIRTY}'" \
-    -output "${DIR}/pkg/{{.OS}}_{{.Arch}}/${TOOL}" \
-    -tags="${BUILD_TAGS}" \
+    -o "bin/${TOOL}" \
+    -tags "${BUILD_TAGS}" \
```
Comment (fairclothjm) [inline scripts/build.sh:50]: If you really want to echo the command being run you could do `set -x` and then `set +x`:

```go
set -x
${GO_CMD} build \
    -gcflags "${GCFLAGS}" \
    -ldflags "-X github.com/hashicorp/${TOOL}/version.GitCommit='${GIT_COMMIT}${GIT_DIRTY}'" \
    -o "bin/${TOOL}" \
    -tags "${BUILD_TAGS}" \
    .
set +x
```

That way you wouldn't have to update the `echo` command everytime the `go build` is updated.

File: scripts/build.sh (hunk 3 above)
Comment (robmonte) [reply scripts/build.sh:44]: Oh thanks for catching that debug statement, I was having an issue with the build working initially but forgot to remove that after fixing the issue

File: acceptance_test.go [hunk 5]
```diff
@@ -14,16 +14,15 @@ import (
 
 const (
 	envVarRunAccTests = "VAULT_ACC"
-	envVarAccessKey   = "ALICLOUD_ACCESS_KEY"
-	envVarSecretKey   = "ALICLOUD_SECRET_KEY"
-
+	envVarAccessKey   = "VAULT_ACC_TEST_ACCESS_KEY_ID"
+	envVarSecretKey   = "VAULT_ACC_TEST_SECRET_KEY"
+	envVarRoleARN     = "VAULT_ACC_TEST_ROLE_ARN"
```
Comment (fairclothjm) [inline acceptance_test.go:19]: nit: should we keep this defined below the comment that describes it? That is usually the pattern I observe for comments on variables

**Review by fairclothjm — APPROVED** [review]

File: acceptance_test.go (hunk 5 above)
Comment (robmonte) [reply acceptance_test.go:19]: I am okay with moving it back down, but I figured such a large comment makes it easy to miss a single variable below it. Up to you.

Comment (fairclothjm) [reply acceptance_test.go:19]: It's just a nit comment so I will leave it up to you 🙂

File: acceptance_test.go [hunk 6]
```diff
@@ -14,16 +14,15 @@ import (
 
 const (
 	envVarRunAccTests = "VAULT_ACC"
-	envVarAccessKey   = "ALICLOUD_ACCESS_KEY"
-	envVarSecretKey   = "ALICLOUD_SECRET_KEY"
-
+	envVarAccessKey   = "VAULT_SECRETS_ACC_TEST_ACCESS_KEY_ID"
```
Comment (fairclothjm) [inline acceptance_test.go:17]: Do we need to update the README as well with these new names?

**Review by fairclothjm — APPROVED** [review]

**Pull request merged**